lib/java/com/google/android/material/divider/MaterialDivider.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright 2020 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.material.divider;
+
+import android.content.Context;
+import android.content.res.TypedArray;
+import android.util.AttributeSet;
+import android.util.TypedValue;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.ViewGroup.LayoutParams;
+
+import androidx.annotation.ColorInt;
+import androidx.annotation.Dimension;
+import androidx.annotation.Px;
+
+import com.google.android.material.R;
+import com.google.android.material.internal.ThemeEnforcement;
+import com.google.android.material.internal.ViewUtils;
+
+import static android.view.ViewGroup.LayoutParams.MATCH_PARENT;
+import static android.view.ViewGroup.LayoutParams.WRAP_CONTENT;
+import static com.google.android.material.theme.overlay.MaterialThemeOverlay.wrap;
+
+public class MaterialDivider extends View {
+
+
+  public MaterialDivider(Context context) {
+    this(context, null);
+  }
+
+  public MaterialDivider(Context context, AttributeSet attrs) {
+    this(context, attrs, R.attr.dividerStyle);
+  }
+
+  boolean isRtlEnabled;
+
+  AttributeSet initialAttrs;
+  int initialDefStyleAttr;
+
+  private static final int DEF_STYLE_RES = R.style.Widget_MaterialComponents_Divider;
+
+  public MaterialDivider(Context context, AttributeSet attrs, int defStyleAttr) {
+
+    super(wrap(context, attrs, defStyleAttr, DEF_STYLE_RES), attrs, defStyleAttr);
+    initialAttrs = attrs;
+    initialDefStyleAttr = defStyleAttr;
+    isRtlEnabled = ViewUtils.isLayoutRtl(this);
+
+    float alpha = attrs.getAttributeFloatValue("http://schemas.android.com/apk/res/android", "alpha", -1);
+    int color = attrs.getAttributeIntValue(android.R.attr.background, -1);
+
+    final TypedArray dividerStyleAttrs = getDividerStyleAttrs();
+    if (alpha == -1) {
+      alpha = dividerStyleAttrs.getFloat(ATTR_ALPHA, -1);
+    }
+    if (color == -1) {
+      color = dividerStyleAttrs.getColor(ATTR_COLOR, -1);
+    }
```

Comment:
The way you're resolving attributes here is not optimal and could be simplified:
```
TypedArray attributes =
        context.obtainStyledAttributes(
            attrs, R.styleable.MaterialDivider, defStyle, DEF_STYLE_RES);
alpha = attributes.getFloat(R.styleable.MaterialDivider_dividerAlpha, 1f);
color =
    MaterialResources.getColorStateList(
        context, attributes, R.styleable.MaterialDivider_dividerColor);
inset = attributes.getDimension(R.styleable.MaterialDivider_dividerInset, 0f);
size = attributes.getDimension(R.styleable.MaterialDivider_dividerSize, 0f);
attributes.recycle();
```